pkg/resources/secret_with_basic_authentication_acceptance_test.go

```diff
@@ -0,0 +1,121 @@
+		Steps: []resource.TestStep{
+			// create
+			{
+				Config: config.FromModel(t, secretModel),
+				Check: resource.ComposeTestCheckFunc(
+					assert.AssertThat(t,
+						resourceassert.SecretWithBasicAuthenticationResource(t, secretModel.ResourceReference()).
+							HasNameString(name).
+							HasDatabaseString(id.DatabaseName()).
+							HasSchemaString(id.SchemaName()).
+							HasUsernameString("foo").
+							HasPasswordString("foo").
+							HasCommentString(""),
+					),
+				),
+			},
+			// set username, password and comment
+			{
+				Config: config.FromModel(t, secretModel.
+					WithUsername("bar").
+					WithPassword("bar").
+					WithComment(comment),
+				),
+				Check: assert.AssertThat(t,
+					resourceassert.SecretWithBasicAuthenticationResource(t, secretModel.ResourceReference()).
+						HasNameString(name).
+						HasDatabaseString(id.DatabaseName()).
+						HasSchemaString(id.SchemaName()).
+						HasUsernameString("bar").
+						HasPasswordString("bar").
+						HasCommentString(comment),
+				),
+			},
+			// import
+			{
+				ResourceName:            secretModel.ResourceReference(),
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"password"},
+				ImportStateCheck: importchecks.ComposeImportStateCheck(
+					importchecks.TestCheckResourceAttrInstanceState(helpers.EncodeResourceIdentifier(id), "name", id.Name()),
+					importchecks.TestCheckResourceAttrInstanceState(helpers.EncodeResourceIdentifier(id), "database", id.DatabaseId().Name()),
+					importchecks.TestCheckResourceAttrInstanceState(helpers.EncodeResourceIdentifier(id), "schema", id.SchemaId().Name()),
+					importchecks.TestCheckResourceAttrInstanceState(helpers.EncodeResourceIdentifier(id), "username", "bar"),
+					importchecks.TestCheckResourceAttrInstanceState(helpers.EncodeResourceIdentifier(id), "comment", comment),
+				),
+			},
+			// unset comment
+			{
+				Config: config.FromModel(t, secretModelWithoutComment),
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						planchecks.ExpectChange(secretModel.ResourceReference(), "comment", tfjson.ActionUpdate, sdk.String(comment), nil),
+					},
+				},
+				Check: assert.AssertThat(t,
+					resourceassert.SecretWithClientCredentialsResource(t, secretModelWithoutComment.ResourceReference()).
+						HasCommentString(""),
+				),
+			},
+			// destroy
+			{
+				Config:  config.FromModel(t, secretModel),
+				Destroy: true,
+			},
+			// create with empty username and password
+			{
+				Config: config.FromModel(t, secretModelEmptyCredentials),
+				Check: resource.ComposeTestCheckFunc(
+					assert.AssertThat(t,
+						resourceassert.SecretWithBasicAuthenticationResource(t, secretModelEmptyCredentials.ResourceReference()).
+							HasNameString(name).
+							HasDatabaseString(id.DatabaseName()).
+							HasSchemaString(id.SchemaName()).
+							HasUsernameString("").
```

Comment:
Hmm that's interesting it doesn't errors out. Can you copy this one case to its dedicated test function where this would be a first case? I have a feeling this should error out when username and password is empty (it's considered as non set field in Terraform and they're marked as Required which seems strange to me it's not an error).
Also, I don't remember, but I'm not sure how `Destroy: true` works in this test suite. I remember it works not as I intended to use it, so maybe it's also causing some confusion here.

Reply:
huh... this is strange, i've removed the `Destroy: true` and moved this case to a separate Test function and still same result, test doesn't fail.

Reply:
I checked in the meantime and this test with `Destroy: true` was good, but I don't understand why it's not failing. I saw it's possible to have empty username and password, so I'm assuming that's why it works.